Sources/protoc-gen-swift/main.swift

```diff
@@ -27,8 +27,10 @@ import PluginLibrary
 enum GenerationError: Error {
   /// Raised for any errors reading the input
   case readFailure
-  /// Raise when parsing the parameter string and found an unknown key
+  /// Raised when parsing the parameter string and found an unknown key
   case unknownParameter(name: String)
+  /// Raised when a parameter was giving an invalid value
+  case invalidParameterValue(name: String, value: String)
```

Comment:
The throwing site we'd still need shim code to transform the enums into an array of strings.  And better docs still seems like the idea solution to this sorta problem.  So I'm leaning towards not doing this at this point and just adding the docs instead.